velox/exec/MergeJoin.cpp

```diff
@@ -1346,18 +1351,29 @@ RowVectorPtr MergeJoin::applyFilter(const RowVectorPtr& output) {
       }
     };
 
+    auto onMatch = [&](auto row, bool& firstMatched_) {
```

Comment:
Let's control the firstMatched_ flag in caller context. Also let's add matching rows here for other join types if possible?